cppForSwig/Makefile.am

```diff
@@ -0,0 +1,110 @@
+LINK = $(CXX)
+
+PYTHON_INCLUDES=@PYTHON_INCLUDES@
+PYTHON_LIBS=@PYTHON_LIBS@
+
+if TARGET_LINUX
+UNAME=Linux
+else
+if TARGET_WINDOWS
+UNAME=Windows
+else
+if TARGET_DARWIN
+UNAME=Darwin
+endif
+endif
+endif
+
+# Export variables to make them available to cryptopp Makefile.
+export AR
+export CXX
+export RANLIB
+export UNAME
+
+CRYPTOPP_EXT=.a
+
+if TARGET_DARWIN
+MACOSX_DEPLOYMENT_TARGET=10.7
+export MACOSX_DEPLOYMENT_TARGET
+LDFLAGS += -undefined dynamic_lookup -headerpad_max_install_names
+endif
+
+OBJS = UniversalTimer.o BinaryData.o lmdb_wrapper.o StoredBlockObj.o \
+	BtcUtils.o BlockObj.o BlockUtils.o EncryptionUtils.o \
+	BtcWallet.o LedgerEntry.o ScrAddrObj.o Blockchain.o BlockWriteBatcher.o \
+	BDM_mainthread.o lmdbpp.o BDM_supportClasses.o \
+	BlockDataViewer.o HistoryPager.o Progress.o \
+	libcryptopp$(CRYPTOPP_EXT) mdb.o midl.o txio.o FileMap.o SSHheaders.o
+
+CPPFLAGS = $(ARMORY_CPPFLAGS) -Icryptopp -Imdb -DUSE_CRYPTOPP -D__STDC_LIMIT_MACROS
+LDLIBS = -lpthread -Lmdb
+SWIG_OPTS    = -c++ -python -threads
+
+SWIG_INC     =
+
+CXXCPP = $(CPPFLAGS) $(PYTHON_INCLUDES)
+
+if TARGET_DARWIN
+CXXCPP += -stdlib=libc++
+LDLIBS += -lc++
+if BUILD_LINUX
+CXXCPP += -I/usr/include/c++/v1
+endif
+endif
+
+if TARGET_LINUX
+LDLIBS += -lrt
+endif
+
+# each .o file depends on all .h files
+EVERY_HEADER = bdmenums.h BDM_mainthread.h BDM_supportClasses.h BinaryData.h \
+	Blockchain.h BlockDataManagerConfig.h BlockDataViewer.h BlockObj.h \
+	BlockUtils.h BlockWriteBatcher.h BtcUtils.h BtcWallet.h EncryptionUtils.h \
+	FileMap.h HistoryPager.h LedgerEntry.h lmdbpp.h lmdb_wrapper.h log.h \
+	OS_TranslatePath.h PartialMerkle.h Progress.h ReorgUpdater.h ScrAddrObj.h \
+	SSHheaders.h StoredBlockObj.h txio.h UniversalTimer.h util.h
+
+
+#**************************************************************************
+
+all: ../_CppBlockUtils.so ../qrc_img_resources.py
```

Comment:
Do you have a link to any more information on targets in ../ causing issues? I'm not doubting you, but I am having trouble finding information about the issues. I'd just like to learn more about it.

I moved the targets formerly known as ../_CppBlockUtils.so ../qrc_img_resources.py into the main Makefile.

I still have the following rule in the cppForSwig Makefile. Is it okay, since it isn't a rule in ../, but just a directory passed to swig?

```
CppBlockUtils_wrap.cxx: $(EVERY_HEADER) CppBlockUtils.i
    $(SWIG) $(SWIG_OPTS) -outdir ../ -v CppBlockUtils.i
```